features/gm.ts

```diff
@@ -299,6 +302,23 @@ export function gmEventType(
     if (receiver == ctx.publicKey.hex) {
         return "gm_invitation"; // received by me
     }
+
+    const isInGroup = Array.from(groupMessageListGetter.getConversationList()).filter((contact) =>
+        contact.pubkey.hex == receiver
+    );
+    if (isInGroup.length > 0) {
+        return "gm_message";
+    }
+
+    const isInConversation = Array.from(conversationListRetriever.getContacts()).filter((contact) =>
+        contact.pubkey.hex == receiver
+    );
+    const isInStrangers = Array.from(conversationListRetriever.getStrangers()).filter((contact) =>
+        contact.pubkey.hex == receiver
+    );
+    if (isInConversation.length > 0 || isInStrangers.length > 0) {
+        return "gm_invitation";
+    }
```

Comment:
This is a very tedious implementation.

If you want to use interfaces, when you should declare the minimal interface this funciton needs which is just a function that returns a boolean to tell if a pubkey is a DM or not.

Reply:
On the other hand, if it's an invitation sent by me, I can always decript it with my own private key. Therefore, there is no need to check what role the p tag is.

Reply:
Either way works, but you should have your own thinking instead of merely following my instructions when bugs such as this arise.